src/TrackGenerator.cpp

```diff
@@ -268,9 +273,27 @@ double TrackGenerator::getZCoord() {
 
 
 /**
+ * @brief Return the array used to store the FSR volumes
+ * @return _FSR_volumes the FSR volumes array indexed by FSR ID
+ */
+FP_PRECISION* TrackGenerator::getFSRVolumesBuffer() {
```

Comment:
I'm not really sure I understand the point of abstracting this code away from `TrackGenerator::getFSRVolumes()`. What was your reasoning?